crates/primitives/src/bytecode.rs

```diff
@@ -66,10 +73,23 @@ impl Bytecode {
         matches!(self, Self::Eof(_))
     }
 
+    /// Checks if the bytecode starts with the EOF prefix followed by the correct version.
+    #[inline]
+    pub fn is_eof_format(bytes: &Bytes) -> bool {
+        bytes.len() >= 2 && bytes[0] == EOF_PREFIX && bytes[1] == EOF_VERSION
+    }
+
     /// Creates a new raw [`Bytecode`].
     #[inline]
     pub fn new_raw(bytecode: Bytes) -> Self {
```

Comment:
Would return Error here as Legacy bytecode with 0xEF is considered invalid. And only valid EOF can have `0xEF` prefix.

Would additionally add `new_legacy` function for legacy codes.

Reply:
Would return error here